app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/step_count/AddStepCount.kt

```diff
@@ -10,11 +10,18 @@ import de.westnordost.streetcomplete.osm.Tags
 class AddStepCount : OsmFilterQuestType<Int>(), AndroidQuest {
 
     override val elementFilter = """
-        ways with highway = steps
-         and (!indoor or indoor = no)
-         and access !~ private|no
-         and (!conveying or conveying = no)
-         and !step_count
+        nodes, ways with
+        (
+            highway = steps
+            and (!indoor or indoor = no)
+            and (!conveying or conveying = no)
+        )
+        or
+        (
+            tower:type = observation
+        )
+        and access !~ private|no
+        and !step_count
```

Comment:
`and` has a higher operator precedence than `or`. This expression will return true for private steps.